p2p/3.IPFS_or_HTTP/musicshare/views.py

```diff
@@ -0,0 +1,56 @@
+import os
+
+from django.http import HttpResponseRedirect, HttpResponse, Http404
+from django.shortcuts import render
+from django.core.paginator import Paginator
+
+from musicshare.forms import SongUploadForm
+from musicshare.models import Song
+
+
+def index(request):
+    songs = Song.objects.all().order_by("id")
+    paginator = Paginator(songs, 8)
+    page_number = request.GET.get('page')
+    page_obj = paginator.get_page(page_number)
+    return render(request, 'musicshare/index.html', {"page_obj": page_obj})
+
+
+def upload(request):
+    if request.method == 'POST':
+        form = SongUploadForm(request.POST, request.FILES)
+        if form.is_valid():
+            '''
+            Step 2.0:
+                Here, you can connect / send a request to ipfs.infura API with "/add" method.
+                The API will respond with the corresponding hash of the file uploaded.
+                Don't forget to create a new Song object and store the hash !
```

Comment:
You could give a link to the Infura Documentation.
I think [this one](https://infura.io/docs/ipfs) is the one you're looking for here.